TwoThousandAndFortyEight.hs

```diff
@@ -1,13 +1,48 @@
 module TwoThousandAndFortyEight where
 
-line :: [Int] -> [String]
-line xs = map show xs 
+data Direction = Left' | Right'
+
+line :: (Num t, Show t) => [t] -> [String]
+line xs = map show xs
+
+-- main :: IO()
+-- main = runProgram drawGrid
+
+runProgram :: (Num t, Show t) => [[t]] -> IO()
+runProgram grid = do
+  putStrLn $ concat (map drawLine $ map line grid)
+  putStr "Do one movement (h, j, k, l): "
+  name <- getLine
+  putStrLn $ "Your movement was: " ++ name
+  runProgram $ map moveAndSquash drawGrid
+
+
+-- @TODO: after grab the movement, should we clean the scream and draw the new board?
+
+drawLine :: [String] -> String
+drawLine (a:b:c:d:_) = " [ " ++ show a ++ " ] [ " ++ show b ++ " ] [ " ++ show c ++ " ] [ " ++ show d ++ " ]\n"
```

Comment:
It looks like this is the same as: ` drawLine = show`!

Reply:
```
Prelude> let ys = [["1"],["2"],["3"],["4"]]
Prelude> show ys
"[[\"1\"],[\"2\"],[\"3\"],[\"4\"]]"
```

Reply:
Don't I need to do a `ys <$> show`? This is the final grid «that will be shown to the users», the `\"` is not really interesting IMO.